shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -45,18 +106,182 @@ impl RedisConfig {
+                                        ];
+
+                                        messages_redis_request.push(Message::from_frame(
+                                            Frame::Redis(RedisFrame::Array(commands_buffer)),
+                                        ));
+                                    }
+                                    Err(err_state) => {
+                                        warn!("build_cache_query err: {}", err_state);
+                                        state = err_state;
+                                    }
+                                }
+                            } else {
+                                state = CacheableState::Skip(format!(
+                                    "Table {} not in caching list",
+                                    table_name
+                                ));
+                            }
+                        } else {
+                            state = CacheableState::Skip(format!(
+                                "{} is not a readable query",
+                                statement
+                            ));
+                        }
+
+                        match state {
+                            CacheableState::Err(_) | CacheableState::Skip(_) => {
+                                debug!("build_cache_query: {}", state);
+                                return Err(state);
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+                _ => {
+                    return Err(CacheableState::Err(format!(
+                        "cannot fetch {cass_request:?} from cache"
+                    )))
+                }
+            }
+        }
+        Ok(messages_redis_request)
+    }
+
+    /// unwraps redis response messages into cassandra messages. It does this by replacing the Cassandra
+    /// request messages with their corresponding Cassandra response messages and returns them.
+    /// Result is either the modified message_cass_request (now response messages) or and CacheableState::Err.
+    fn unwrap_cache_response(
+        &self,
+        messages_redis_response: Messages,
+        mut cassandra_messages: Messages,
+    ) -> Result<Messages, CacheableState> {
+        // Replace cass_request messages with cassandra responses in place.
+        // We reuse the vec like this to save allocations.
+        let mut messages_redis_response_iter = messages_redis_response.into_iter();
+        /* there is a redis response for each statement in a CassandraMessage so we have to map
+        the redis responses back to the cassandra requests
+         */
+
+        for cass_request in cassandra_messages.iter_mut() {
+            // the responses for this request
+            let cassandra_result: Result<CassandraResult, CacheableState> =
+                if let Some(Frame::Cassandra(frame)) = &mut cass_request.frame() {
```

Comment:
This can be just `if let Some(Frame::Cassandra(frame)) = cass_request.frame() {` because `frame()` already returns an `&mut`